client/state/data-layer/wpcom/videos/poster/index.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * External dependencies
+ */
+import { pick } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { VIDEO_EDITOR_UPDATE_POSTER } from 'state/action-types';
+import {
+	setPosterUrl,
+	showError,
+	showUploadProgress,
+} from 'state/ui/editor/video-editor/actions';
+
+/**
+ * Updates the poster for a video.
+ *
+ * @param  {Object} store  Redux store
+ * @param  {Object} action  Action object
+ * @param {Function} next  Dispatches to next middleware in chain
+ * @returns {Object} original action
+ */
+export const updatePoster = ( { dispatch }, action, next ) => {
+	if ( ! ( 'file' in action.params || 'atTime' in action.params ) ) {
+		return next( action );
+	}
+
+	const { atTime, file } = pick( action.params, [ 'atTime', 'file' ] );
+	const params = Object.assign(
+		{
+			apiVersion: '1.1',
+			method: 'POST',
+			path: `/videos/${ action.videoId }/poster`,
+		},
+		file && { formData: [ [ 'poster', file ] ] },
+		atTime !== undefined && { body: { at_time: atTime } },
+	);
+
+	dispatch( http( params, action ) );
+
+	return next( action );
+};
+
+export const updatePosterUrl = ( { dispatch }, action, next, { poster } ) => {
+	dispatch( setPosterUrl( poster ) );
+};
+
+export const updatePosterError = ( { dispatch } ) => {
+	dispatch( showError() );
```

Comment:
this error concept is still bugging me. we don't actually show an error do we? it seems like all we really do is indicate that the poster is no longer updating. is the error coming later?

is there another way to represent the poster URL which would indicate if it's reset or set? on error, instead of sending an error action, would it make sense to revert the URL to its previous state, or an empty string? if so, we could instead dispatch something like `setPosterUrl( previousURL )`